alf/environments/suite_carla.py

```diff
@@ -936,13 +996,26 @@ def reset(self):
 
         self._fail_frame = None
         self._done = False
-        p = np.array([loc.x, loc.y, loc.z])
         self._prev_location = loc
         self._prev_action = np.zeros(
             self.action_spec().shape, dtype=np.float32)
         self._alf_world.update_actor_location(self._actor.id, loc)
 
-        self._navigation.set_destination(goal_loc)
+        self._route_length = self._navigation.set_destination(goal_loc)
+
+        self._prev_collision = False  # whether there is collision in the previous frame
+        self._collision = False  # whether there is colliion in the current frame
+        self._collision_loc = None  # the location of the car when it starts to have collition
+
+        # The intermediate goal for sparse reward
+        self._intermediate_goal_index = min(self._sparse_reward_distance,
+                                            self._navigation.num_waypoints - 1)
```

Comment:
It might be possible that the waypoints are not evenly spaced in the future. Carla leaderboard does use non-evenly spaced waypoints. So index is a proper name.

Reply:
If waypoints may not be evenly spaced, then what does this line mean min(sparse_reward_distance, _num_waypoints -1)?

Reply:
I agree that there is some confusion here. sparse_reward_distance is meant to be distance and intermediate_goal_index is meant to be index. Currently, the distances between two adjacent waypoints are approximately 1 meter (this can not be true for every adjacent pairs because the route length is not guaranteed to be integer number of meters). Updated code to use sampling_resolution to calculating the index.